gossip/src/duplicate_shred_handler.rs

```diff
@@ -63,10 +60,17 @@ pub struct DuplicateShredHandler {
     // We don't want bad guys to inflate the chunk map, so we limit the number of
     // pending proofs from each pubkey to ALLOWED_SLOTS_PER_PUBKEY.
     validator_pending_proof_map: HashMap<Pubkey, HashSet<Slot>>,
-    // remember the last root slot handled, clear anything older than last_root.
-    last_root: Slot,
+    // Cache last root to reduce read lock.
+    last_root: Option<Slot>,
+    // remember the last root slot cleaned, clear anything between last_root and last_root_cleaned.
+    last_root_cleaned: Option<Slot>,
```

Comment:
Theoretically because root could potentially be 0, but I guess it never happens in production, and it's not catastrophic if we don't handle duplicate shreds when root is 0, so I'm changing last_root and last_root_cleaned to 0 now.

Reply:
Why can't they get actual values at initialization?